de-lib/src/main/java/org/iplantc/de/client/services/impl/DiskResourceServiceFacadeImpl.java

```diff
@@ -1041,6 +1042,48 @@ public String downloadTemplate(String templateid) {
         return address;
     }
 
+    @Override
+    public void requestHTPathlistFile(HTPathListRequest request, DECallback<File> callback) {
+        StringBuilder address =
+                new StringBuilder(deProperties.getDataMgmtBaseUrl() + "path-list-creator?");
+        address.append("dest=" + URL.encodeQueryString(request.getDest()));
+        if(!Strings.isNullOrEmpty(request.getPattern())) {
+            address.append("&name-pattern=" + URL.encodeQueryString(request.getPattern()));
+        }
+        address.append("&folders-only=" + request.isFoldersOnly());
+        address.append("&recursive=" + true);
+        if (request.getInfoTypes() != null && request.getInfoTypes().size() > 0) {
+            for (String infotype : request.getInfoTypes()) {
+                address.append("&info-type=" + infotype);
+            }
+        }
+        Splittable s = AutoBeanCodex.encode(AutoBeanUtils.getAutoBean(request));
+        Splittable spath = StringQuoter.createSplittable();
+        s.get("paths").assign(spath,"paths");
```

Comment:
So I am just making another change to ` HTPathListRequest`  to extend `HasPaths`. Even with that I don't see how I could HasPaths AutoBean to send paths in the request body. I need to convert it `Splittable` so that I can place the path array under `paths` key.